controllers/recipeController.js

```diff
@@ -346,13 +346,19 @@ recipeController.updateRecipe = (req, res) => {
             res.sendStatus(500);
         }
     } else {
-        db.query(recipeObject.upsertData(recipeData), (err) => {
-            if (err) {
-                res.sendStatus(500);
-            } else {
-                res.sendStatus(200);
-            }
-        });
+        if(!recipeData.fieldName || !recipeData.value){
+            res.sendStatus(500);
```

Comment:
404 if request is empty

Reply:
we have in controller method getRecipeById in which we have (!id.match(/^[0-9]+$/) condition where checking  id with string method